### PR TITLE
Add C# 9

### LINKED ISSUE #20352: Outdated
[Enter feedback here]
Era 2017 describing future specs. Now we ARE 2020 and C# 8 is concrete reality. Update links!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a60b6e17-36cf-bd6c-1b76-d38f1760a816
* Version Independent ID: 0fdeaa81-c3c2-c290-224e-792fcee9b297
* Content: [C# reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/)
* Content Source: [docs/csharp/language-reference/index.md](https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/index.md)
* Product: **dotnet-csharp**
* Technology: **csharp-language-reference**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #20352

### CODE REVIEW

File: docs/csharp/language-reference/index.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 description: "C# reference"
 title: "C# reference"
-ms.date: 02/14/2017
+ms.date: 01/13/2021
 f1_keywords: 
   - _CSharpKeyword
 helpviewer_keywords:
```
Comment (Youssef1313) [inline docs/csharp/language-reference/index.md:7]: Consider adding `ms.custom: "updateeachrelease"` to the metadata.

File: docs/csharp/language-reference/index.md [hunk 2]
```diff
@@ -55,7 +55,10 @@ The features that have been implemented in C# versions after 6.0 are represented
   
  [C# 8.0 Specification Proposals](../../../_csharplang/proposals/csharp-8.0/nullable-reference-types.md)  
  C# 8.0 is available with .NET Core 3.0. The features include nullable reference types, recursive pattern matching, default interface methods, async streams, ranges and indexes, pattern based using and using declarations, null coalescing assignment, and readonly instance members.
-  
+
+ [C# 9.0 Specification Proposals](../../../_csharplang/proposals/csharp-9.0/records.md)  
+ C# 9.0 is available with .NET 5.0. The features include records, top-level statements, nullable reference types, pattern matching enhancements, init only setters, target-typed new expressions, module initializers, extending partial methods, static anonymous functions, target-typed conditional expressions, covariant return types, extension GetEnumerator in foreach loops, lambda discard parameters, attributes on local functions, native sized integers, function pointers, suppress emitting localsinit flag, and unconstrained type parameter annotations.
```
Comment (BillWagner) [inline docs/csharp/language-reference/index.md:60]: Nullable Reference types was in C# 8:

```suggestion
 C# 9.0 is available with .NET 5.0. The features include records, top-level statements, pattern matching enhancements, init only setters, target-typed new expressions, module initializers, extending partial methods, static anonymous functions, target-typed conditional expressions, covariant return types, extension GetEnumerator in foreach loops, lambda discard parameters, attributes on local functions, native sized integers, function pointers, suppress emitting localsinit flag, and unconstrained type parameter annotations.
```

**Review by BillWagner — APPROVED** [review]
Thanks @tdykstra 

I had one suggestion, and then this is ready to :shipit:

File: docs/csharp/language-reference/index.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 description: "C# reference"
 title: "C# reference"
-ms.date: 02/14/2017
+ms.date: 01/13/2021
 f1_keywords:
```
Comment (tdykstra) [inline docs/csharp/language-reference/index.md:5]:
```suggestion
ms.date: 01/13/2021
ms.custom: "updateeachrelease"
f1_keywords: 
```

File: docs/csharp/language-reference/index.md (hunk 2 above)
Comment (tdykstra) [reply docs/csharp/language-reference/index.md]: Nullable ref types spec is in the C# 9 TOC node -- should it not be there?
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/nullable-reference-types-specification

Comment (BillWagner) [reply docs/csharp/language-reference/index.md]: Mads updated the spec, but the behavior is (mostly) the same. 

Sadly, this is a point in time problem will the ECMA committee catches up.

**Pull request merged**